src/eval/execution.rs

```diff
@@ -14,15 +14,43 @@ static INIT_V8: Once = Once::new();
 
 pub fn evaluate(nix_expr: &str) -> EvalResult {
     initialize_v8();
+    // Declare the V8 execution context
     let isolate = &mut v8::Isolate::new(Default::default());
     let scope = &mut v8::HandleScope::new(isolate);
     let context = v8::Context::new(scope);
     let scope = &mut v8::ContextScope::new(scope, context);
+    let global = context.global(scope);
 
+    let import_module_attr = v8::String::new(scope, "importNixModule").unwrap();
+    let import_module_fn = v8::Function::new(scope, import_nix_module).unwrap();
+    global
+        .set(scope, import_module_attr.into(), import_module_fn.into())
+        .unwrap();
+
+    // Execute the Nix runtime JS module, get its exports
+    let nixjs_rt_str = include_str!("../../nixjs-rt/dist/lib.mjs");
+    let nixjs_rt_obj = exec_module(nixjs_rt_str, scope).unwrap();
+
+    // Set them to a global variable
+    let nixrt_attr = v8::String::new(scope, "n").unwrap();
+    global
+        .set(scope, nixrt_attr.into(), nixjs_rt_obj.into())
+        .unwrap();
+
+    let root_nix_fn = eval_nix_fn_from_string(scope, nix_expr)?;
+
+    nix_value_from_module(scope, root_nix_fn, nixjs_rt_obj)
+}
+
+fn eval_nix_fn_from_string<'s>(
+    scope: &mut HandleScope<'s>,
+    nix_expr: &str,
+) -> Result<v8::Local<'s, v8::Function>, String> {
     let source_str = emit_module(nix_expr)?;
-    let source_v8 = to_v8_source(scope, &source_str, "<eval string>");
-    let module = v8::script_compiler::compile_module(scope, source_v8).unwrap();
+    let module_source_v8 = to_v8_source(scope, &source_str, "<eval string>");
+    let module = v8::script_compiler::compile_module(scope, module_source_v8).unwrap();
```

Comment:
For another day: It would be great to use some kind of caching here (to avoid re-emitting JS or even recompiling JS&mdash;with [v8's code-caching ability](https://v8.dev/blog/code-caching-for-devs)).